Docs: Add docs for new default auth UI customization feature (#30830)
hashicorp/vault: website/content/api-docs/system/config-ui-login-default-auth.mdx

---
layout: api
page_title: /sys/config/ui/login/default-auth - HTTP API
description: The '/sys/config/ui/login/default-auth' endpoint configures default authentication types that will display on a UI page per namespace.
---

# `/sys/config/ui/login/default-auth`

@include 'alerts/enterprise-only.mdx'

> @include 'alerts/restricted-admin.mdx'

Author: Added the new line here!

---
layout: api
page_title: /sys/config/ui/login/default-auth - HTTP API
description: The '/sys/config/ui/login/default-auth' endpoint configures default authentication types that will display on a UI page per namespace.
---

# `/sys/config/ui/login/default-auth`

@include 'alerts/enterprise-only.mdx'

@include 'alerts/restricted-admin.mdx'

The `/sys/config/ui/login/default-auth` endpoint is used to list, create, read, update, or delete login configurations for the Vault GUI. Setting this configuration customizes the Vault UI login form and determines which methods to display in the login form.

> ## Create UI default login configuration

Reviewer: ```suggestion
## Create GUI default login configuration
```

> This endpoint allows the creation of a new UI default login customization configuration. Only one UI default login configuration is

Reviewer 2: I think `default` can be removed here since that's implied by the title and first sentence?
```suggestion
This endpoint allows the creation of a new UI default login customization configuration. Only one UI login configuration is
```

allowed per namespace. Namespaces can inherit from their parent if the parent's configuration has `disable_inheritance`
set to `false`. If no namespace is provided, the rule will be applied to the root namespace. One of `default_auth_type`,
or `backup_auth_types` is required during creation.

| Method | Path                              |
| :----- | :-------------------------------- |
| `POST` | `/sys/config/ui/login/default-auth/:name` |

### Parameters

- `name` `(string: <required>)` - The name of the configuration. Spaces are not supported. This parameter is specified as part of the URL.

> - `namespace_path` `(string: "")` - Namespace to apply this configuration to.

Reviewer: ```suggestion
- `namespace_path` `(string: "")` - Target namespace for the login configuration.
  Leave `namespace_path` unset to apply the configuration to the `root` namespace.
```
Style correction: avoid "this" as a pronoun, write in active voice

> - `default_auth_type` `(string: <required>)` - The type of auth method to be presented by default. Only UI supported auth types are allowed: `github`,`jwt`,`ldap`,`oidc`,`okta`,`radius`,`saml`,token`,and `userpass`

Reviewer 2: ```suggestion
- `default_auth_type` `(string: <required>)` - The type of auth method to be presented by default. Only UI supported auth types are allowed: `github`,`jwt`,`ldap`,`oidc`,`okta`,`radius`,`saml`, `token`,and `userpass`
```

- `backup_auth_types` `(array: [])` - Backup auth types to be presented in the `Sign in with other methods` tab. Only UI supported auth types are allowed: `github`,`jwt`,`ldap`,`oidc`,`okta`,`radius`,`saml`, `token`,and `userpass`

> - `disable_inheritance` `(bool: false)` - Defaults to `false` which means child namespaces inherit `default_auth_type` and `backup_auth_types from their parent. If set to `true`, the config will not be inherited.

Reviewer 2: ```suggestion
- `disable_inheritance` `(bool: false)` - Defaults to `false` which means child namespaces inherit `default_auth_type` and `backup_auth_types` from their parent. If set to `true`, the config will not be inherited.
```

### Sample payload

```json
{
  "namespace_path": "it_admins",
  "default_auth_type": "okta",
  "backup_auth_types": ["userpass", "token"],
  "disable_inheritance": "true"
}
```

### Sample request

```shell-session
$ curl --header "X-Vault-Token: ..." \
    --request POST \
    --data @payload.json
    http://127.0.0.1:8200/v1/sys/config/ui/login/default-auth/it-admin-rule
```

> ## List UI default login configurations

Reviewer: ```suggestion
## List GUI default login configurations
```

This endpoint returns a list of all UI default login configurations that exist.

| Method | Path                              |
| :----- | :-------------------------------- |
| `LIST` | `/sys/config/ui/login/default-auth` |

### Sample request

```shell-session
$ curl --header "X-Vault-Token: ..." \
    --request LIST \
    http://127.0.0.1:8200/v1/sys/config/ui/login/default-auth
```

### Sample response

```json
{
  "data": {
    "key_info": {
      "default-ldap": {
        "name":"default-ldap",
        "namespace_path":"root",
        "disable_inheritance":false
      }
    },
    "keys":["default-ldap"]
  }
}
```

> ## Read UI default auth configuration

Reviewer: ```suggestion
## Read GUI default auth configuration
```

This endpoint returns the properties of a specific UI default auth configuration.

| Method |                                      |
| :----- | :----------------------------------- |
| `GET`  | `/sys/config/ui/login/default-auth/:name` |

### Parameters

> - `name` `(string: <required>)` - The name of the configuration

Reviewer: ```suggestion
- `name` `(string: <required>)` - Path parameter indicating the name of login
  configuration to fetch.
```

### Sample request

```shell-session
$ curl --header "X-Vault-Token: ..." \
    http://127.0.0.1:8200/v1/sys/config/ui/login/default-auth/it-admin-rule
```

### Sample response

```json
{
  "data": {
    "backup_auth_types":["token","userpass"],
    "default_auth_type":"okta",
    "disable_inheritance":false,
    "namespace_path":"it_admins"
  }
}
```

> ## Update UI default auth configuration

Reviewer: ```suggestion
## Update GUI default auth configuration
```

> This endpoint updates the properties of a specific UI default auth configuration.

Reviewer: ```suggestion
Update the properties of a specific GUI default auth configuration.
```

| Method |                                      |
| :----- | :----------------------------------- |
| `POST` | `/sys/config/ui/login/default-auth/:name` |

### Parameters

> - `namespace_path` `(string: "")` - Namespace to apply this configuration to.

Reviewer: ```suggestion
- `name` `(string: <required>)` - Path parameter indicating the name of the
  configuration. Names can contain letters, numbers, underscores, and dashes.

- `namespace_path` `(string: "")` - Target namespace for the login configuration.
  Leave `namespace_path` unset to update configurations under the `root`
  namespace.
```

`name` should be defined here as well, no?

> - `default_auth_type` `(string: <required>)` - The type of auth method to be presented by default. Only UI supported auth types are allowed: `github`,`jwt`,`ldap`,`oidc`,`okta`,`radius`,`saml`,`token`,and `userpass`

Reviewer: ```suggestion
- `default_auth_type` `(string: <required>)` - The default authentication method.
  You must provide a default method if `backup_auth_types` is unset. Supported
  authentication methods include: `github`, `jwt`, `ldap`, `oidc`, `okta`,
  `radius`, `saml`, `token`, and `userpass`.
```

Edited to mirror create endpoint

> - `backup_auth_types` `(array: [])` - Backup auth types to be presented in the `Sign in with other methods` tab. Only UI supported auth types are allowed: `github`,`jwt`,`ldap`,`oidc`,`okta`,`radius`,`saml`, `token`,and `userpass`

Reviewer: ```suggestion
- `backup_auth_types` `(array: [])` - A list of backup auth types. Vault
  presents the backup methods in the `Sign in with other methods` tab. You must
  provide at least one backup method if `default_auth_types` is unset. Supported
  authentication methods include: `github`, `jwt`, `ldap`, `oidc`, `okta`,
  `radius`, `saml`, `token`, and `userpass`.
```
Edited to mirror create endpoint

> - `disable_inheritance` `(bool: false)` - Disable child namespaces from inheriting `default_auth_type` and `backup_auth_types` from this configuration.

Reviewer: ```suggestion
- `disable_inheritance` `(bool: false)` - Indicates whether child namespaces
  should inherit `default_auth_type` and `backup_auth_types` settings from their
  parent. Setting `disable_inheritance` to `true` disallows inheritence.
```
Edited to mirror create endpoint

### Sample payload

```json
{
  "namespace_path": "it_admins",
  "default_auth_type": "ldap",
  "backup_auth_types": ["userpass", "github"],
  "disable_inheritance": "true"
}
```

### Sample request

```shell-session
$ curl --header "X-Vault-Token: ..." \
    --request POST \
    --data @payload.json
    http://127.0.0.1:8200/v1/sys/config/ui/login/default-auth/it-admin-rule
```

> ## Delete UI default auth configuration

Reviewer: ```suggestion
## Delete GUI default auth configuration
```

> This endpoint deletes a specific UI default auth configuration.

Reviewer: ```suggestion
Delete the target GUI auth configuration.
```

| Method   |                                      |
| :------- | :----------------------------------- |
| `DELETE` | `/sys/config/ui/login/default-auth/:name` |

### Parameters

> - `name` `(string: <required>)` - The name of the configuration

Reviewer: ```suggestion
- `name` `(string: <required>)` - Path parameter indicating the name of the
  configuration. Names can contain letters, numbers, underscores, and dashes.
```
